_posts/2020-10-30-snowflake-dataOps.md

```diff
@@ -0,0 +1,165 @@
+Like DevOps, DataOps minimises manual interventions by codifying data related changes and deployment through automated CI/CD tools.
+<br/>
+In this blog post, I will show you how to create your Snowflake DataOps pipeline using AWS developer tools and Flyway.
+
+<div style="background-color: #fff3cd ; border-color: #ffeeba; color: #856404; border-radius: .25rem; padding: .75rem 1.25rem;"><strong>Warning!</strong><br/>This is a very hands-on blog post. Get ready to get your hands dirty &#128525;</div> <br/>
+
+## Let's get Started:
+
+To get started you will need to follow below steps:
+
+# Step 1: Downloading the template
+Download below repository as a zip folder. Once downloaded unzip it and cd into the folder<br/>
+<a href="https://github.com/MechanicalRock/snowflake-dataops">Snowflake Dataops Repository</a>
+
+This repo uses below services to implement an automated deployment cycle to Snowflake. All the pipeline and related infrastructure are created through code using AWS cloudformation.
+
+* CodeCommit
+* CodeBuild
+* CodePipeline
+* Cloudformation
+* Flyway
+
+# Step 2: Setup Snowflake Credentials
+
+To allow your pipeline to get access to Snowflake, you will need to first create a RSA public and private key. You can create the keys using openssl by running below commands in your command line
+```
+openssl genrsa 2048 | openssl pkcs8 -topk8 -inform PEM -out keys/rsa_key.p8 -nocrypt
+openssl rsa -in keys/rsa_key.p8 -pubout -out keys/rsa_key.pub
+```
+Once keys are created, you then need to create a Snowflake user and assign the RSA public key to it. <i>Run below commands on Snowflake worksheet or execute them using snowsql</i>
+
+Note: Remove/exclude the header and footer of the public key
+```
+create user pipeline_sys_user;
+alter user pipeline_sys_user set rsa_public_key_2='MIIBIjANBgkqh...';
+```
+
+Next, in order for your pipeline to use the private key, you must store it in AWS secrets manager. 
+
+Store the private key in aws secrets manager as plain text and remove/exclude the header and footer of the private key
+<center><img src="/img/blog/dataops/secretsmanager-1.png" /></center><br/>
+Make sure to name it "snowflake/pipeline_sys_user"
+<center><img src="/img/blog/dataops/secretsmanager-2.png" /></center><br/>
+
+Last step is allowing code build in your pipeline to access the key in secrets manager.
+To do that find out the arn of your secrets manager:
+<center><img src="/img/blog/dataops/secretsmanager-3.png" /></center><br/>
+Open aws_seed.yaml file and update line number 404 with the arn to your secret manager password
+```
+  - <The arn to the secrets manager that holds Snowflake Password>
```

Comment:
couldn't this be a parameter in the cloudformation template?

Reply:
Because the secret is stored manually it can't be used as a export parameter

Reply:
I'll change it to pass it as parameter but they still have to enter the arn manually